docs/advanced/experimental/kubernetes.md

```diff
@@ -1,20 +1,68 @@
-# Jina on Kubernetes
+# Jina On Kubernetes
 
 Deploy your `Flow` on `Kubernetes`.
 ## Requirements
-Please setup a `Kubernetes` cluster and pull the credentials. e.g.:
+Please setup a `Kubernetes` cluster and pull the credentials.
 
--  [minikube](https://minikube.sigs.k8s.io/docs/start/) (testing),
+For local testing `minikube` is recommended.
+- [minikube](https://minikube.sigs.k8s.io/docs/start/), 
+  
+Here are some managed `Kubernetes` cluster solutions you could use:
 - [Google Kubernetes Engine](https://cloud.google.com/kubernetes-engine),
 - [Amazon EKS](https://aws.amazon.com/eks),
 - [Azure Kubernetes Service](https://azure.microsoft.com/en-us/services/kubernetes-service),
 - [Digital Ocean](https://www.digitalocean.com/products/kubernetes/)
   
-## Deploy your `Flow`
+## Deploy Your `Flow`
 To deploy a `Flow` on `Kubernetes`, you have to set the `infrastructure='K8S'` when creating the `Flow`.
-Call `flow.start()` to trigger the deployment. Make sure all `Executors` are containerized.
+```{caution}
+All Executors in the Flow should be used with `jinahub+docker://`.
+```
+
+## Example 1 - CLIP Image Encoder
+The following code deploys a simple `Flow` with just one `Executor`.
+```python
+from jina import Flow
+
+flow = Flow(
+    name='test-flow', port_expose=8080, infrastructure='K8S', protocol='http'
+).add(
+    uses='jinahub+docker://CLIPImageEncoder',
+)
+flow.start()
+
+```
+After the deployment finished, run a port forward to run client requests.
+```bash
+kubectl port-forward svc/gateway -n index-flow 8080:8080 &
+```
+
+Once the port forward setup, you can request embeddings for images.
+```python
+import base64
 
-## Example
+import numpy as np
+import requests
+from jina import Document
+
+host = '127.0.0.1'
+port = 8080
```

Comment:
Maybe add a comment that this is `port_expose` from above

Reply:
thanks, added